vllm/entrypoints/cli/serve.py

```diff
@@ -162,7 +166,20 @@ def run_multi_api_server(args: argparse.Namespace):
 
     assert not args.headless
     num_api_servers = args.api_server_count
-    # assert num_api_servers > 1
+
+    assert num_api_servers > 1
+    if "PROMETHEUS_MULTIPROC_DIR" not in os.environ:
+        # Make TemporaryDirectory for prometheus multiprocessing
+        # Note: global TemporaryDirectory will be automatically
+        #   cleaned up upon exit.
+        global prometheus_multiproc_dir
+        prometheus_multiproc_dir = tempfile.TemporaryDirectory()
+        os.environ["PROMETHEUS_MULTIPROC_DIR"] = prometheus_multiproc_dir.name
+    else:
+        logger.warning("Found PROMETHEUS_MULTIPROC_DIR was set by user. "
+                       "This directory must be wiped between vLLM runs or "
+                       "you will find inaccurate metrics. Unset the variable "
+                       "and vLLM will properly handle cleanup.")
```

Comment:
We should probably only do this is prometheus logging is disabled .. at least we probably shouldn't if `log_stats` is `False`.

Reply:
Setting the env var even if log_stats is false is fine? it doesn't hurt? Why do you think we should gate it on log_stats?  or even to be more precise when `Prometheus` logger is used?